whois-api/src/main/java/net/ripe/db/whois/api/rdap/RdapObjectMapper.java

```diff
@@ -656,38 +658,27 @@ private static AsBlockRange getAsBlockRange(final String asBlock) {
         }
     }
 
-    private static void handleLanguageAttribute(final RpslObject rpslObject, final RdapObject rdapObject) {
-        final List<RpslAttribute> languages = rpslObject.findAttributes(AttributeType.LANGUAGE);
-        if (languages.isEmpty()) {
-            return;
-        }
-        // TODO: [ES] add warning notice if additional languages found
-        rdapObject.setLang(languages.get(0).getCleanValue().toString());
-        addNoticeForMultipleValues(rdapObject, AttributeType.LANGUAGE, languages, rpslObject.getKey().toString());
-    }
-
-    private static void handleCountryAttribute(final RpslObject rpslObject, final Ip ip) {
-        final List<RpslAttribute> countries = rpslObject.findAttributes(AttributeType.COUNTRY);
-        if (countries.isEmpty()) {
-            return;
-        }
-        // TODO: [ES] add warning notice if additional languages found
-        ip.setCountry(countries.get(0).getCleanValue().toString());
-        addNoticeForMultipleValues(ip, AttributeType.COUNTRY, countries, ip.getHandle());
-    }
-
-    private static void addNoticeForMultipleValues(final RdapObject rdapObject, final AttributeType type, final List<RpslAttribute> values, final String key) {
-        if (values.isEmpty() || values.size() == 1) {
-            return;
-        }
-
-        final String commaSeperatedValues = values.stream().map(RpslAttribute::getCleanValue).collect(Collectors.joining(", "));
-        final String title = String.format("Multiple %s attributes found", type.getName());
-        final String desc = String.format("There are multiple %s attributes %s in %s, but only the first %s %s was returned.", type.getName(), commaSeperatedValues, key, type.getName(), values.get(0).getCleanValue());
-
-        final Notice notice = new Notice();
-        notice.setTitle(title);
-        notice.getDescription().add(desc);
-        rdapObject.getNotices().add(notice);
+    private static void handlePartialValueAttributes(final RpslObject rpslObject, final RdapObject rdapObject){
```

Comment:
Does this logic need to be combined into one method? The term "partial value" is not defined. Can language and country handling be kept separate?

Reply:
I think so, if not both method will have exactly the same logic. The difference will be the attribute...which for me makes it better to keep in just one method (same logic - just different attributes)
On the other hand, if you keep it into different methods you will need to call it from 

createIp -> To the one related to the country and lang
createEntity -> To the one that just create the country

So you will need to do the call from different parts and to do exactly the same logic.

About the name I am totally agree that can be better, I accept suggestions I will try to think about a better name.

Reply:
"handleMultipleValuesAttributes" this could be a better name because we already know that in the RDAP class object we just can have a key-value attributes. So this name I think that makes clear that we are handling those attributes with multiple values